routes/userRoutes.js

```diff
@@ -33,4 +34,40 @@ userRouter.get('/id=:id', authorizedOrdinaryUser, async (req, res) => {
     }
 });
 
+userRouter.get("/candidateVoted/id=:id", authorizedOrdinaryUser, async (req, res) => {
+    const userId = req.params.id;
+    if(userId === undefined || userId === null || userId < 0) {
+        logger.error(`User id ${userId} is invalid!`);
+        return res.status(400).send({ error: "User id is invalid!" });
+    }
+    try {
+        logger.debug("Getting user by id: ", userId);
+        const user = await User.find({ id: Number(userId) });
+        if (user.length === 1) {
+            const candidateVotedId = user[0].candidateVotedId;
+            if(candidateVotedId === undefined || candidateVotedId === null || candidateVotedId < 0) {
+                logger.error(`Candidate voted id ${candidateVotedId} is invalid!`);
+                return res.status(204).send({ });
+            }
+            const candidate = await Candidate.find({ id: Number(candidateVotedId) });
+            if (candidate.length === 1) {
+                const result = {
+                    id: candidate[0].id,
+                    name: candidate[0].name,
+                    message: candidate[0].message,
+                    img: candidate[0].img
+                };
+                logger.info("Candidate found successfully! Details:", result);
+                return res.status(200).send(result);
+            }
+            return res.status(204).send({ });
+        }
+        logger.error("Couldn't find user with id: ", userId);
+        return res.status(400).send({ error: "User Not Found!" });
```

Comment:
Should this be 404 or 400

Reply:
https://stackoverflow.com/questions/34312023/http-get-request-status-204-vs-404

Reply:
I'm asking for the status code if the user is not found, but according to the link shouldn't it be 204 instead of 400?

Reply:
Oh yeah, sounds right :smile: